chore : use --no-cache-dir flag to pip in dockerfiles to save space

rajpratik71 opened the pull request:
<!--- changelog-entry
# Fill in 'msg' below to have an entry automatically added to the next release changelog.
# Leaving 'msg' blank will not generate a changelog entry for this PR.
# Please ensure this is a simple (and readable) one-line string.
---
msg: "chore : use --no-cache-dir flag to pip in dockerfiles to save space"
-->

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

using --no-cache-dir flag in pip install ,make sure downloaded packages
by pip don't cached on system . This is a best practice which make sure
to fetch from repo instead of using local cached one . Further , in case
of Docker Containers , by restricting caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

rajpratik71 committed:
chore : use --no-cache-dir flag to pip in dockerfiles to save space

using --no-cache-dir flag in pip install ,make sure downloaded packages
by pip don't cached on system . This is a best practice which make sure
to fetch from repo instead of using local cached one . Further , in case
of Docker Containers , by restricting caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

fosterseth commented:
we could consider adding this flag to the requirements.txt install as well

https://github.com/ansible/awx/blob/8126f734e37f25ed523ad9a36f61d2e98e65fa60/Makefile#L114

rajpratik71 committed:
chore : use --no-cache-dir flag to pip in dockerfiles to save space

using --no-cache-dir flag in pip install ,make sure downloaded packages
by pip don't cached on system . This is a best practice which make sure
to fetch from repo instead of using local cached one . Further , in case
of Docker Containers , by restricting caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

rajpratik71 commented:
> we could consider adding this flag to the requirements.txt install as well
> 
> https://github.com/ansible/awx/blob/8126f734e37f25ed523ad9a36f61d2e98e65fa60/Makefile#L114

@fosterseth  added as requested . Please approve the CI.

shanemcd commented:
Sorry for the delay here. It looks like there is now a conflict and there needs to be a rebase.

rajpratik71 committed:
Merge branch 'devel' into optimization/pip-no-cache-dir

rajpratik71 commented:
> Sorry for the delay here. It looks like there is now a conflict and there needs to be a rebase.

done , lets wait for CI

john-westcott-iv commented:
Thanks for your contribution. We have added this to our backlog to testing tested and merged. It will need another rebase but we can wait until an engineer is assigned to the work. In the mean time, please feel free to reach out to us on the #ansible-awx Matrix channel if you would like to discuss this PR.

rajpratik71 committed:
Merge branch 'devel' into optimization/pip-no-cache-dir

TheRealHaoLiu commented:
CI stuck, so i closed and reopen the PR
